src/main/java/org/iot/dsa/dslink/dynamodb/DynamoDBNode.java

```diff
@@ -0,0 +1,469 @@
+        act.setResultType(ResultType.VALUES);
+        act.addColumnMetadata(Constants.RESULT, DSValueType.STRING);
+        return act;
+    }
+
+    private DSAction makeUpdateItemDynamoDBAction() {
+        DSAction act = new DSAction.Parameterless() {
+            @Override
+            public ActionResult invoke(DSInfo info, ActionInvocation invocation) {
+                return ((DynamoDBNode) info.get()).updateItem(this,invocation.getParameters());
+            }
+        };
+        act.addParameter(Constants.TABLENAME, DSValueType.STRING, "Table Name");
+        act.addParameter(Constants.PRIMARYKEY, DSValueType.STRING, "Primary Key");
+        act.addParameter(Constants.UPDATEEXPRESSION, DSValueType.STRING, "UpdateExpression");
+        act.addParameter(Constants.CONDITIONEXPRESSION, DSValueType.STRING, "ConditionExpression");
+        act.addParameter(Constants.EXPRESSIONATTRIBUTENAMES, DSValueType.STRING, "Expression Attribute Names");
+        act.addParameter(Constants.EXPRESSIONATTRIBUTEVALUES, DSValueType.STRING, "Expression Attribute Values");
+        act.setResultType(ResultType.VALUES);
+        act.addColumnMetadata(Constants.RESULT, DSValueType.STRING);
+        return act;
+    }
+
+    private DSAction makeDeleteItemDynamoDBAction() {
+        DSAction act = new DSAction.Parameterless() {
+            @Override
+            public ActionResult invoke(DSInfo info, ActionInvocation invocation) {
+                return ((DynamoDBNode) info.get()).deleteItem(this,invocation.getParameters());
+            }
+        };
+        // Add parameters as needed
+        act.addParameter(Constants.TABLENAME, DSValueType.STRING, "Table Name");
+        act.addParameter(Constants.PRIMARYKEY, DSValueType.STRING, "Primary Key");
+        act.addParameter(Constants.CONDITIONEXPRESSION, DSValueType.STRING, "ConditionExpression");
+        act.addParameter(Constants.EXPRESSIONATTRIBUTENAMES, DSValueType.STRING, "Expression Attribute Names");
+        act.addParameter(Constants.EXPRESSIONATTRIBUTEVALUES, DSValueType.STRING, "Expression Attribute Values");
+        act.setResultType(ResultType.VALUES);
+        act.addColumnMetadata(Constants.RESULT, DSValueType.STRING);
+        return act;
+    }
+
+    ///////////////////////////////////////////////////////////////////////////
+    // Actions
+    ///////////////////////////////////////////////////////////////////////////
+
+    /**
+     * Edit Dynamodb Settings
+     */
+    private void editDynamoDB(DSMap parameters){
+        setParameters(parameters);
+        debug("Account Edited" + getInfo().getName());
+        init();
+    }
+
+    /**
+     * Query Dynamodb
+     */
+    private ActionResult queryDynamoDB(DSAction action,DSMap parameters){
+        DSActionValues actionResult = new DSActionValues(action);
+        try {
+            DSMap queryResult = client.QueryItem(parameters);
+            actionResult.addResult(DSString.valueOf(new String(queryResult.toString())));
```

Comment:
Returning the JSON of the result from dynamoDB works for now, but it would be better to return a table, with a row for each item.

Reply:
We thought to make it as close as to DynamoDB. As the data is no SQL format, it wont have fix number of fields for each record. So thought to provide data in raw format. if needed customer can use our JSON parser logic-block to get table if they need. But we can create table format if needed.